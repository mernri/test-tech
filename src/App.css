.App {
  color: black;
  background-color: #f6f7fb;
  min-height: 100vh;
}

/* message-form  */

#message-form {
  display: flex;
  flex-direction: column;
  width: 70vw;
  margin: 0vh auto 5vh auto;
  padding-top: 2em;
}

#message-form h2 {
  margin-bottom: 0;
}

form {
  display: flex;
  flex-direction: column;
}

form #content {
  margin-top: 1em;
  margin-bottom: 1em;
  height: 5vh;
  box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.1);
  border-radius: 5px;
  border-width: 0px;
  padding-left: 1em;
  padding-top: 0em;
}

form #radio-visibility {
  margin-left: 0px;
  margin-bottom: 1em;
}

form label {
  padding-right: 1em;
}

form button {
  height: 5vh;
  background-color: #2196f3;
  box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.1);
  border-radius: 5px;
  color: white;
  font-size: 1em;
  font-weight: 600;
}

hr {
  width: 70vw;
  margin: auto;
  border-top: 1px solid #f1f1f1;
}

/* message-list  */

#message-list {
  width: 70vw;
  display: flex;
  flex-direction: column;
  margin: auto;
}

#message-list h2 {
  margin-bottom: 0;
}

.messages ul {
  list-style: none;
  padding-left: 0px;
}

.messages ul li {
  display: flex;
  flex-direction: row;
  padding: 5px 20px;
  background-color: #ffffff;
  box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.1);
  border-radius: 5px;
}

.messages .visibility {
  margin-left: auto;
  color: gray;
}
