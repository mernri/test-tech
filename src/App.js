import React from "react";
import MessageForm from "./components/MessageForm.js";
import MessageList from "./components/MessageList.js";
import axios from "axios";
import "./App.css";

// TO LAUNCH THE FAKE API : // npm run mock-api

class App extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      messages: []
    };
  }

  componentDidMount() {
    this.getAllMessages();
  }

  // Retrieves all the messsages
  getAllMessages = () => {
    let url = "http://localhost:3001/messages";
    axios
      .get(url)
      .then(response => {
        this.setState({
          // .reverse() => reverses the order of the elements in the array
          messages: response.data.reverse()
        });
      })
      .catch(err => {
        console.log(err);
      });
  };

  // Adds the new message to the list of messages
  addMessage = async (content, visibility) => {
    let url = "http://localhost:3001/messages";
    await axios.post(url, { content: content, visibility: visibility });
    this.getAllMessages();
  };

  render() {
    return (
      <div className="App">
        <div id="message-form">
          <h2>Publier un message</h2>
          <MessageForm addMessage={this.addMessage} />
        </div>
        <hr />
        <div id="message-list">
          <h2>Liste des messages publi√©s</h2>
          <MessageList messages={this.state.messages} />
        </div>
      </div>
    );
  }
}

export default App;
