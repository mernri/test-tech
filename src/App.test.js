import React from "react";
import ReactDOM from "react-dom";
import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import App from "./App";

Enzyme.configure({ adapter: new Adapter() });

/* 
Unit test: Testing one isolated function, or one React component. 
Enzyme’s shallow() is a unit test.

Integration test: Testing a multitude of functions working together, 
or an entire React component including children components. 
Enzyme’s mount() is an integration test.

Mock function: Redefining a function specifically for a test to generate a result. 
E.g. returning hard-coded data instead of relying on fetch requests or database calls. 
This strategy could prevent the hard-coded sum issue we were discussing earlier!
Mock functions can be defined in jest with jest.fn(() => { //function here });
*/

describe("Testing App Compoment with Enzyme", () => {
  it("renders without crashing", () => {
    const component = shallow(<App />);
    expect(component).toMatchSnapshot()
  });
});
