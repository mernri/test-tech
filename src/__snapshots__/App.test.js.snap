// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing App Compoment with Enzyme renders without crashing 1`] = `ShallowWrapper {}`;
